consensus/src/quorum_store/proof_manager.rs

```diff
@@ -88,23 +226,67 @@ impl ProofManager {
                     PayloadFilter::InQuorumStore(proofs) => proofs,
                 };
 
-                let proof_block = self.proofs_for_consensus.pull_proofs(
-                    &excluded_batches,
-                    max_txns,
-                    max_bytes,
-                    return_non_full,
+                let (proof_block, proof_queue_not_fully_utilized) = self
```

Comment:
nit: can you change the semantic to return if the proof queue is fully utilized rather than the negation? That way you can get rid of negations in the code block below and improve the readability.

Reply:
Done.